src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -3518,5 +3519,23 @@ public void PropertyFunctionEnsureTrailingSlash()
 
             Assert.Equal(path + Path.DirectorySeparatorChar, result);
         }
+
+        [Fact]
+        public void PropertyFunctionWithNewLines()
+        {
+            const string propertyFunction = @"$(SomeProperty
+.Substring(0, 10)
+.Substring(0, 5))";
```

Comment:
Does this test hit all 3 modified lines? If not, can you add cases until it does?